pkg/cmd/issue/develop/develop.go

```diff
@@ -21,225 +20,164 @@ import (
 type DevelopOptions struct {
 	HttpClient func() (*http.Client, error)
 	GitClient  *git.Client
-	Config     func() (config.Config, error)
 	IO         *iostreams.IOStreams
 	BaseRepo   func() (ghrepo.Interface, error)
 	Remotes    func() (context.Remotes, error)
 
-	IssueRepoSelector string
-	IssueSelector     string
-	Name              string
-	BaseBranch        string
-	Checkout          bool
-	List              bool
+	IssueSelector string
+	Name          string
+	BaseBranch    string
+	Checkout      bool
+	List          bool
 }
 
 func NewCmdDevelop(f *cmdutil.Factory, runF func(*DevelopOptions) error) *cobra.Command {
 	opts := &DevelopOptions{
 		IO:         f.IOStreams,
 		HttpClient: f.HttpClient,
 		GitClient:  f.GitClient,
-		Config:     f.Config,
 		BaseRepo:   f.BaseRepo,
 		Remotes:    f.Remotes,
 	}
 
 	cmd := &cobra.Command{
-		Use:   "develop [flags] {<number> | <url>}",
+		Use:   "develop {<number> | <url>}",
 		Short: "Manage linked branches for an issue",
 		Example: heredoc.Doc(`
-			$ gh issue develop --list 123 # list branches for issue 123
-			$ gh issue develop --list --issue-repo "github/cli" 123 # list branches for issue 123 in repo "github/cli"
-			$ gh issue develop --list https://github.com/github/cli/issues/123 # list branches for issue 123 in repo "github/cli"
-			$ gh issue develop 123 --name "my-branch" --base my-feature # create a branch for issue 123 based on the my-feature branch
-			$ gh issue develop 123 --checkout # fetch and checkout the branch for issue 123 after creating it
+			# List branches for issue 123
+			$ gh issue develop --list 123
+
+			# List branches for issue 123 in repo cli/cli
+			$ gh issue develop --list --repo cli/cli 123
+
+			# Create a branch for issue 123 based on the my-feature branch
+			$ gh issue develop 123 --base my-feature
+
+			# Create a branch for issue 123 and checkout it out
+			$ gh issue develop 123 --checkout
 			`),
 		Args: cmdutil.ExactArgs(1, "issue number or url is required"),
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			// This is all a hack to not break the issue-repo flag. It will be removed
+			// in the near future and this hack can be removed at the same time.
+			flags := cmd.Flags()
+			if flags.Changed("issue-repo") && !flags.Changed("repo") {
+				repo, _ := flags.GetString("issue-repo")
+				_ = flags.Set("repo", repo)
+			}
+			if cmd.Parent() != nil {
+				return cmd.Parent().PersistentPreRunE(cmd, args)
+			}
+			return nil
```

Comment:
This basically changes `--issue-repo` to use `--repo` underneath the covers.

Reply:
my understanding was that issue-repo was functionally and meaningfully different from --repo since this command can span repositories.

`gh issue develop -Rvilmibm/cli --name 123-bugfix --base some-feature --issue-repo cli/cli 123`

in this invocation I'm saying to create the branch `123-bugfix` based on the branch `some-feature` in the repository `vilmibm/cli` and link it to the issue numbered 123 in the `cli/cli` repo.

Reply:
Huh... well that I definitely didn't think about that. It also was not clear from the documentation at all. I will add back in the `--issue-repo` functionality and update the documentation to better reflect its purpose and how it differs from the `--repo` flag.

Reply:
This is still very confusing behavior IMO. I would much rather there be a `--branch-repo` flag instead of an `--issue-repo` flag. That would better match the behavior of the other `issue` commands.